Interpolate clip

manuelsh opened the pull request:
This PR addresses the suggestions of @amyeroberts in the existing PR https://github.com/huggingface/transformers/pull/30783.

nileshkokane01 committed:
fixes clip interpolate

nileshkokane01 committed:
fixes clip interpolate

nileshkokane01 committed:
fixes clip interpolate

Commit:
All suggestions have been addressed, plus some changes done by the `make repo-consistency` and related processes.

amyeroberts commented on tests/models/x_clip/test_modeling_x_clip.py:
```diff
@@ -731,3 +731,25 @@ def test_inference(self):
+        # to visualize self-attention on higher resolution images.
+        model = XCLIPModel.from_pretrained("microsoft/xclip-base-patch32").to(torch_device)
+
+        image_processor = XCLIPProcessor.from_pretrained("microsoft/xclip-base-patch32", size=480)
```
The returned object is a processor - not an image processor 

```suggestion
        processor = XCLIPProcessor.from_pretrained("microsoft/xclip-base-patch32", size=480)
```

amyeroberts commented on tests/models/x_clip/test_modeling_x_clip.py:
```diff
@@ -731,3 +731,25 @@ def test_inference(self):
+        inputs = image_processor(text="what's in the video", videos=video, return_tensors="pt").to(torch_device)
+
+        # forward pass
+        with torch.no_grad():
+            outputs = model(**inputs, interpolate_pos_encoding=True)
```
Can you add in the test to check an error is raise if `nterpolate_pos_encoding=False`?

amyeroberts commented on tests/models/kosmos2/test_modeling_kosmos2.py:
```diff
@@ -762,3 +762,27 @@ def test_snowman_image_captioning_batch(self):
+
+        # forward pass
+        with torch.no_grad():
+            outputs = model(**inputs, interpolate_pos_encoding=True)
```
Same here re failing

manuelsh replied:
Hy @amyeroberts , I am getting the same results in `model` with  `interpolate_pos_encoding=True` or `False` in both x_clip or kosmos2. No error, just the same exact tensor for `outputs.vision_model_output.last_hidden_state`.

manuelsh replied:
I think the reason is because a logic like this:

``` if not interpolate_pos_encoding and (height != self.image_size[0] or width != self.image_size[1]):
            raise ValueError(
                f"Input image size ({height}*{width}) doesn't match model"
                f" ({self.image_size[0]}*{self.image_size[1]})."
            )
```
is not implemented in the `modeling_kosmos2.py` file and its corresponding one in `modeling_x_clip.py`. 

Happy to implement it.

manuelsh replied:
Part of the issue is originated, in the case of the model kosmos2, because 

```
        processor = AutoProcessor.from_pretrained(
            "microsoft/kosmos-2-patch14-224", padding_side="left", size={"shortest_edge": 480}
        )
```
is not changing the image size. It is still returning a 224x224 image, even when you reduce the shortest_edge to less than 224.

Need to investigate further.

manuelsh replied:
I've found out what was missing:
```
        # default imnage size of pretrained kosmos_2 is 224 x 224
        processor = AutoProcessor.from_pretrained(
            "microsoft/kosmos-2-patch14-224", size={"shortest_edge": 180}, crop_size = {'height': 180, 'width': 180}
        )
```

Now, I have implemented the ValueError in the `modeling_kosmos2.py` file.

amyeroberts reviewed (commented):
Thanks for adding this! 

Just a small comment on the tests

Commit:
adding use case of interpolate_pos = False and respective testing to all models

manuelsh commented:
@amyeroberts I've implemented changes in all clip family models to account for the use case when `interpolate_pos_encoding=False` and the resolution is not the same as the pretrained model. I also implemented  its respective tests.

Commit:
Merge branch 'interpolate_clip' of https://github.com/nileshkokane01/transformers into interpolate_clip

Commit:
merging conflicts

Commit:
formatting

Commit:
formatting

manuelsh commented:
I've attempted to sync with the huggingface:main branch, as there are some failed tests due to being out of sync.

Commit:
syncing with main branch

manuelsh commented:
Closing this one in favour of #32600
